src/Storage/Services/ApplicationService.cs

```diff
@@ -38,5 +41,34 @@ public ApplicationService(IApplicationRepository applicationRepository)
 
             return (false, new ServiceError(405, $"DataType {dataType} is not declared in application metadata for app {appId}"));
         }
+
+        /// <summary>
+        /// Wrapper method for getting an application or an error message using the application id.
+        /// </summary>
+        /// <param name="appId">The application id</param>
+        /// <returns></returns>
+        public async Task<(Application Application, ServiceError ServiceError)> GetApplicationOrErrorAsync(string appId)
+        {
+            ServiceError serviceError = null;
+            Application appInfo = null;
+
+            try
+            {
+                string org = appId.Split("/")[0];
+                appInfo = await _applicationRepository.FindOne(appId, org);
+
+                if (appInfo == null)
+                {
+                    serviceError = new ServiceError(404, $"Did not find application with appId={appId}");
+                }
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "An error occurred while fetching application with appId={appId}", appId);
```

Comment:
We can use PascalCasing as suggested:
```suggestion
                _logger.LogError(e, "An error occurred while fetching application with appId={AppId}", appId);
```